tests/subsys/fs/common/test_fs_open_flags.c

```diff
@@ -0,0 +1,313 @@
+
+/* C preprocessor redefinitions that automatically fill in line parameter */
+#define ZOPEN(ts, flags, expected) ZOPEN(ts, flags, expected, __LINE__)
+#define ZCLOSE(ts) ZCLOSE(ts, __LINE__)
+#define ZWRITE(ts, expected) ZWRITE(ts, expected, __LINE__)
+#define ZREAD(ts, expected) ZREAD(ts, expected, __LINE__)
+#define ZUNLINK(ts) ZUNLINK(ts, __LINE__)
+#define ZCHKPOS(ts, expected) ZCHKPOS(ts, expected, __LINE__)
+#define ZREWIND(ts) ZREWIND(ts, __LINE__)
+
+/* Create empty file */
+#define ZMKEMPTY(ts)			\
+do {					\
+	ZUNLINK(ts);			\
+	ZOPEN(ts, FS_O_CREATE, 0);	\
+	ZCLOSE(ts);			\
+} while (0)
+
+void test_fs_open_flags(void)
+{
+	struct test_state ts = {
+		*&test_fs_open_flags_file_path,
+		{ 0 },
+		something,
+		RDWR_SIZE,
+		buffer,
+		RDWR_SIZE,
+	};
+	int block = 1;
+
+	ZBEGIN("Attempt open non-existent");
+	ZOPEN(&ts, 0, -ENOENT);
+	ZOPEN(&ts, FS_O_WRITE, -ENOENT);
+	ZOPEN(&ts, FS_O_READ, -ENOENT);
+	ZOPEN(&ts, FS_O_RDWR, -ENOENT);
+	ZOPEN(&ts, FS_O_APPEND, -ENOENT);
+	ZOPEN(&ts, FS_O_APPEND | FS_O_READ, -ENOENT);
+	ZOPEN(&ts, FS_O_APPEND | FS_O_WRITE, -ENOENT);
+	ZOPEN(&ts, FS_O_APPEND | FS_O_RDWR, -ENOENT);
+	ZEND();
+
+
+	/* Attempt create new file with no read/write access and check
+	 * operations on it.
+	 */
+	ZBEGIN("Attempt create new with no R/W access");
+	ZOPEN(&ts, FS_O_CREATE | 0, 0);
+	ZWRITE(&ts, -EACCES);
+	ZREAD(&ts, -EACCES);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt create new with READ access");
+	ZOPEN(&ts, FS_O_CREATE | FS_O_READ, 0);
+	ZWRITE(&ts, -EACCES);
+	ZREAD(&ts, 0);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt create new with WRITE access");
+	ZOPEN(&ts, FS_O_CREATE | FS_O_WRITE, 0);
+	ZWRITE(&ts, ts.write_size);
+	ZREAD(&ts, -EACCES);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt create new with R/W access");
+	ZOPEN(&ts, FS_O_CREATE | FS_O_RDWR, 0);
+	ZWRITE(&ts, ts.write_size);
+	/* Read is done at the end of file, so 0 bytes will be read */
+	ZREAD(&ts, 0);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt open existing with no R/W access");
+	ZMKEMPTY(&ts);
+	ZOPEN(&ts, 0,  0);
+	ZWRITE(&ts, -EACCES);
+	ZREAD(&ts, -EACCES);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt open existing with READ access");
+	ZMKEMPTY(&ts);
+	ZOPEN(&ts, FS_O_READ,  0);
+	ZWRITE(&ts, -EACCES);
+	/* File is empty */
+	ZREAD(&ts, 0);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt open existing with WRITE access");
+	ZMKEMPTY(&ts);
+	ZOPEN(&ts, FS_O_WRITE,  0);
+	ZWRITE(&ts, ts.write_size);
+	ZREAD(&ts, -EACCES);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt open existing with R/W access");
+	ZMKEMPTY(&ts);
+	ZOPEN(&ts, FS_O_RDWR,  0);
+	ZWRITE(&ts, ts.write_size);
+	/* Read is done at the end of file, so 0 bytes will be read */
+	ZREAD(&ts, 0);
+	ZCLOSE(&ts);
+	ZUNLINK(&ts);
+	ZEND();
+
+
+	ZBEGIN("Attempt append existing with no R/W access");
+	ZMKEMPTY(&ts);
+	ZOPEN(&ts, FS_O_APPEND,  0);
```

Comment:
Add:

    ZCHKPOS(&ts, 0);

Also after all other opens that include `FS_O_APPEND`.

Reply:
Done.